skimage/measure/fit.py

```diff
@@ -837,7 +834,11 @@ def ransac(data, model_class, min_samples, residual_threshold,
     # estimate model for current random sample set
     model = model_class()
 
-    for num_trials in range(max_trials):
+    num_trials = 0
+
+    while num_trials < max_trials:
+        num_trials += 1
+
```

Comment:
```suggestion
    num_trials = 0
    # max_trials can be updated inside the loop, so this cannot be a for-loop
    while num_trials < max_trials:
        num_trials += 1
```

I found this confusing when I first read it and was about to ask you to revert to a for-loop, so I think leaving a comment here for future readers will be useful.